### PR TITLE
Lybecker/compositionservicesdocs

### PR DESCRIPTION
**Pull request by Lybecker**
# PR for issue #335 

## What is being addressed

- Describes the overall architecture
- Describes in details the Composition Service and the flow of the Composition Service.

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -1,3 +1,82 @@
-# Network Architecture
+# Overview
 
-![Architecture](./assets/architecture.png)
+The Azure Trusted Research Environment (TRE) consists of multiple components all encapsulated in networks with restricted in- & egress traffic. One network for management components and one network per Workspace. All traffic has to explicit allowed by the Application Gateway of the Firewall. Azure TRE is build around the concept of data in a Workspace must not be able to exfiltrate the Workspace.
```
Comment (christoferlof) [inline docs/architecture.md:3]:
```suggestion
The Azure Trusted Research Environment (TRE) consists of multiple components, all encapsulated in networks with restricted ingress- & egress traffic. There is one network for the management components and one network per Workspace. All traffic has to be explicitly allowed by the Application Gateway or the Firewall.
```

File: docs/architecture.md [hunk 2]
```diff
@@ -1,3 +1,82 @@
...
+
+![Architecture overview](./assets/archtecture-overview.png)
```
Comment (christoferlof) [inline docs/architecture.md:5]: I suggest we move as much as possible of the descriptive text out of the image and into the doc instead. Also, let's see if we can align the boxes in the diagram.

File: docs/architecture.md [hunk 3]
```diff
@@ -1,3 +1,82 @@
...
+
+Management consists of two groups of components.
```
Comment (christoferlof) [inline docs/architecture.md:7]:
```suggestion
The Azure TRE management plane consists of two groups of components:
```

**Review by tompaana — APPROVED** [review]

File: docs/architecture.md [hunk 4]
```diff
@@ -1,3 +1,82 @@
...
+
+- Management API & Composition Service
+- Shared Services (not implemented yet [#23](https://github.com/microsoft/AzureTRE/issues/23), [#22](https://github.com/microsoft/AzureTRE/issues/21), & [#21](https://github.com/microsoft/AzureTRE/issues/21))
```
Comment (christoferlof) [inline docs/architecture.md:10]:
```suggestion
- Shared Services

> Shared Services is still work in progress. Please see [#23](https://github.com/microsoft/AzureTRE/issues/23), [#22](https://github.com/microsoft/AzureTRE/issues/21), & [#21](https://github.com/microsoft/AzureTRE/issues/21)
```

File: docs/architecture.md [hunk 5]
```diff
@@ -1,3 +1,82 @@
...
+A Workspace is an instance of a Workspace Template. A Workspace Template is implemented as a [Porter](https://porter.sh/) bundle - read more about [Authoring Workspaces Templates](authoring-workspaces.md). A Porter bundle is a full encapsulated bundle with everything needed (binaries, scripts, IoC templates etc.) to provision an instance of Workspace Template.
+The [TRE Administrator](./user-roles.md#tre-administrator) can register a Porter bundle to use the Composition Service to provision instances of the Workspace Templates.
+To do so requires:
+
+1. The Porter bundle to be pushed to the Azure Container Registry (ACR).
+1. Registering the Workspace through the Management API.
+
+### Provision a Workspace
+
+![Composition Service](./assets/composition-service.png)
```
Comment (christoferlof) [inline docs/architecture.md:32]: Same as for the diagram above - We should move text out of the diagram for readability.

**Review by christoferlof — APPROVED** [review]

File: docs/architecture.md (hunk 2 above)
Comment (Lybecker) [reply docs/architecture.md:5]: Removed most of the text as suggested, except descriptions of elements

File: docs/architecture.md (hunk 5 above)
Comment (Lybecker) [reply docs/architecture.md:34]: Removed the text and enlarged the number reference icons

**Pull request merged**